src/ast/mapper.ts

```diff
@@ -11,6 +11,27 @@ import { ASTVisitor } from './visitor';
 */
 
 export class ASTMapper extends ASTVisitor<void> {
+  // List of passes that should have been run before this one
+  prerequisite: Set<string> = new Set<string>();
+
+  addPassPrerequisite(pass_key: string) {
```

Comment:
you can make the logic easier and change the argument from to `pass_key : string` to `passKeys : Set<string>` and then just do `prerequisites.add(...[...passKeys])`

Reply:
My intent was that we can add a single key to pre-req list for a Pass through addPassPrerequisite function which can be used later. Also, unpacking like this is not supported in add method for a set.